src/external/ACSAdapter/AMSFileManager.ts

```diff
@@ -307,10 +322,26 @@ class AMSFileManager {
                 return undefined;
             }
 
-            const {view_location} = response;
-
             let blob: any;  // eslint-disable-line @typescript-eslint/no-explicit-any
 
+            const {view_location, scan} = response;
+
+            if (this.options.fileScan?.disabled === false && scan && scan?.status !== AMSViewScanStatus.PASSED) {

```

Comment:
when status is in progress, it's going to throw an exception??

Reply:
No, it just adds the file to be in the `AMSFileScanner`. The scanner would periodically validate the file until it changes state.